From f79f6cffe3f1d0395ecebb95611563b4179a1d5d Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:38:04 -0400
Subject: [ethernet] i40e: disable FCoE for MFP modes

Message-id: <1409658042-22689-124-git-send-email-sassmann@redhat.com>
Patchwork-id: 90324
O-Subject: [RHEL7.1 PATCH 123/281] i40: disable FCoE for MFP modes
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Vasu Dev <vasu.dev@intel.com>

Currently FCoE is not supported with MFP modes, so this patch
makes sure it is disabled.

It is disabled by overriding HW FCoE capability, so that later
it can be re-enabled by just the SW upgrade.

Change-ID: I1c0bae5c099b209f56b88bda360031a8565e43e8
Signed-off-by: Vasu Dev <vasu.dev@intel.com>
Tested-By: Jack Morgan <jack.morgan@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 566bb85db73969453791c05d7fbd83d488247b65)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_common.c b/drivers/net/ethernet/intel/i40e/i40e_common.c
index 064b0aa..2329e2f 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_common.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_common.c
@@ -1949,6 +1949,12 @@ static void i40e_parse_discover_capabilities(struct i40e_hw *hw, void *buff,
   }
  }
 
+ /* Software override ensuring FCoE is disabled if npar or mfp
+  * mode because it is not supported in these modes.
+  */
+ if (p->npar_enable || p->mfp_mode_1)
+  p->fcoe = false;
+
  /* additional HW specific goodies that might
   * someday be HW version specific
   */
-- 
1.7.1